### DISCUSSION TITLE
bug: Not allowing to limit Azure AD authentication for selected users in same tenent

### DISCUSSION THREAD

**Discussion by sathinduga** [langfuse/langfuse#1385] [Ideas]
### Describe the bug

In self-hosted Langfuse, once we disable the username and password login and use Azure AD (now called Entra ID) authentication it's allowing users to use our AD authentication. 

But the issue is it's allowing every user in the same tenent to login. Even though we limit the Azure AD application to certain users, other users can sign up/log in. So auth scope is not limited to the AD application users.

Following are the environment variables to set. We need `NEXT_PUBLIC_SIGN_UP_DISABLED` as false because otherwise it wouldn't allow new AD users to sign up (even the one we included in AD app).
```
NEXT_PUBLIC_SIGN_UP_DISABLED=false
AUTH_DISABLE_USERNAME_PASSWORD=true
```

```
AUTH_AZURE_AD_CLIENT_ID
AUTH_AZURE_AD_CLIENT_SECRET
AUTH_AZURE_AD_TENANT_ID
```

### To reproduce

Set above environment variables and use an AD user which is not included in the Azure AD application but in the same tenent. User will be able to sign up / sign in.

### Additional information

NextAuth refer this for the same kind of implementations - [https://next-auth.js.org/configuration/callbacks](https://next-auth.js.org/configuration/callbacks )

**Comment by marcklingen** [comment]
Hi @SathinduGA, thanks for raising this! As this is an advanced enterprise requirement, we are open for PRs or to chat about a commercial relationship to prioritize this. You can reach out via marc at langfuse dot com

**Comment by foragerr** [comment]
I may be in a similar situation. To start with no one could login using Azure AD, thanks for your tip about setting `NEXT_PUBLIC_SIGN_UP_DISABLED=false` @SathinduGA.

In your situation, how are uses that aren't assigned in AzureAD able to login?

**Reply by marcklingen** [reply]
Thanks for sharing, very much open for PRs on this. We just use the NextAuth/Auth.js and expose the configuration options for AzureAD. I'd assume that it is possible to restrict which users are able to login via the AzureID configs, at least that's possible for Google OAuth Apps.
